### PR TITLE
Adding song model

### PR DESCRIPTION
**Pull request by erys**
There are two todos in song model, one for COLORS which is currently duplicated in both song and photo models and should only be in one once the first one gets committed. The other todo is for unlock conditions, which I will implement later.

### CODE REVIEW

**Comment by erys** [comment]
Changed the second todo: Adding unlock chapter for songs unlocked by story now, doing other unlock methods later

File: majilove/models.py [hunk 1]
```diff
@@ -106,3 +108,80 @@ def display_blood_type(self):
...
+
+
+############################################################
+# Songs
+
+class Song(MagiModel):
+    collection_name = 'song'
+    owner = models.ForeignKey(User, related_name='added_songs')
+
+    DIFFICULTY_VALIDATORS = [
```
Comment (db0company) [inline majilove/models.py]: Can you move it to before the details of the difficulties?

File: majilove/models.py [hunk 2]
```diff
@@ -106,3 +108,80 @@ def display_blood_type(self):
...
+    lyricist = models.CharField(_('Lyricist'), max_length=100, null=True)
+    LYRICISTS_CHOICES = LANGUAGES_NEED_OWN_NAME
+    d_lyricists = models.TextField(_('Lyricist'), null=True)
+    arranger = models.CharField(_('Arranger'), max_length=100, null=True)
+    ARRANGERS_CHOICES = LANGUAGES_NEED_OWN_NAME
+    d_arrangers = models.TextField(_('Arranger'), null=True)
+
+    singers = models.ManyToManyField(Idol, related_name="sung_songs", verbose_name=_('Singers'))
+
+    # TODO: whichever of photo or song goes in second should take this from the other
```
Comment (db0company) [inline majilove/models.py]: I don't understand this? 🤔  You're saying when we add a song, we should take the color from the photo? Which photo? I don't see any foreign key between the 2.

Comment (db0company) [reply majilove/models.py]: Oh, I think you mean we should just do: `COLOR_CHOICES = Photo.COLOR_CHOICES`, is that correct?

Comment (db0company) [reply majilove/models.py]: Anyway you don't need `COLOR_WITHOUT_I_CHOICES` and you can use string keys (same as comments from Photo PR).

Comment (erys) [reply majilove/models.py]: yeah, that's what I meant in the comment

**Comment by erys** [comment]
Made changes you asked for.

**Review by db0company — APPROVED** [review]
🚀

**Pull request merged**